### PR TITLE
regex: enable safe handling of invalid UTF-8 by default

### PR DESCRIPTION
**Pull request by vtjnash**

### CODE REVIEW

File: base/pcre.jl [hunk 1]
```diff
@@ -48,45 +48,72 @@ end
...
+      UCP               |
       UNGREEDY          |
-      UTF               |
-      UCP
-
+      USE_OFFSET_LIMIT  |
+      UTF
+
+# arguments to pcre2_set_newline
+const COMPILE_NEWLINE_MASK = (
```
Comment (JeffBezanson) [inline base/pcre.jl:84]: What is this for?

**Comment by JeffBezanson** [comment]
Any performance cost?

File: base/pcre.jl (hunk 1 above)
Comment (vtjnash) [reply base/pcre.jl:84]: Seemed like someone wanted them in a list, so I kept them here

**Comment by vtjnash** [comment]
We have some really good @nanosoldier `runbenchmarks("shootout" || "problem", vs=":master")` for those 😬

**Comment by nanosoldier** [comment]
Something went wrong when running [your job](https://github.com/JuliaLang/julia/pull/39524#issuecomment-774276167):
```
ProcessExitedException(2)
```
cc @christopher-dG

**Comment by vtjnash** [comment]
@nanosoldier `runbenchmarks("shootout" || "problem", vs=":master")`

**Comment by nanosoldier** [comment]
[Your benchmark job](https://github.com/JuliaLang/julia/pull/39524#issuecomment-786421948) has completed - possible performance regressions were detected. A full report can be found [here](https://github.com/JuliaCI/NanosoldierReports/blob/master/benchmark/by_hash/87d7368_vs_b39d697/report.md). cc @christopher-dG

**Comment by vtjnash** [comment]
Seems safe to say this is well within the noise, as the benchmarks that include a regex explicitly (as opposed to using them from somewhere else, like Base.Filesystem) are:
```
src/problem/IMDBGraphs.jl
src/problem/JSONParse.jl
src/problem/SpellCheck.jl
src/shootout/k_nucleotide.jl
src/shootout/regex_dna.jl
src/shootout/revcomp.jl
```

**Comment by vtjnash** [comment]
Will merge soon, unless any objections

**Comment by StefanKarpinski** [comment]
I object.

**Comment by vtjnash** [comment]
Ah, sorry, I assumed you were in favor of this from past discussions. What's your objection?

**Comment by StefanKarpinski** [comment]
This is an improvement but my understanding is that this treats `\x80` in a regex literal as U+80 whereas I think it should either be an error or match the byte 0x80.

**Comment by vtjnash** [comment]
No, you're thinking of the other issue. This doesn't change the match string interpretation. This simply sets the flag to disable UB in pcre2 when the searched string isn't valid utf8.

**Pull request merged**